static/templates/actors/spellcasting-dialog.hbs

```diff
@@ -34,48 +27,30 @@
                 <div class="form-group">
                     <label>{{localize "PF2E.ProficiencyLabel"}}</label>
                     <select name="system.proficiency.slug">
-                        {{#select system.proficiency.slug}}
-                            <option value="">{{localize "PF2E.Actor.Creature.Spellcasting.Label"}}</option>
-                            {{#each statistics as |statistic|}}
-                                <option value="{{statistic.slug}}">{{statistic.label}}</option>
-                            {{/each}}
-                        {{/select}}
+                        {{selectOptions statistics selected=system.proficiency.slug blank="PF2E.Actor.Creature.Spellcasting.Label" valueAttr="slug" labelAttr="label" localize=true}}
                     </select>
                 </div>
             {{/if}}
             <div class="form-group">
                 <label>{{localize "PF2E.MagicTraditionLabel"}}</label>
                 <select name="system.tradition.value">
-                    {{#select system.tradition.value}}
-                        {{#if (eq system.prepared.value "items")}}
-                            <option value="">{{localize "PF2E.MagicTraditionUseSpellLabel"}}</option>
-                        {{/if}}
-                        {{#each magicTraditions as |label mode|}}
-                            <option value="{{mode}}">{{localize label}}</option>
-                        {{/each}}
-                    {{/select}}
+                    {{#if (eq system.prepared.value "items")}}
+                        <option value="">{{localize "PF2E.MagicTraditionUseSpellLabel"}}</option>
+                    {{/if}}
+                    {{selectOptions magicTraditions selected=system.tradition.value localize=true}}
```

Comment:
I'm a bit concerned about this one. If selected is blank, it won't select the blank one. Maybe it'll work incidentally because of how a re-render works, Blank can't be contextfual I imagine? What if false is passed to blank?

Reply:
Oh I missed that. Yeah, that probably needs handling in `getData` I'll get to that tomorrow.

Reply:
Maybe its ok to have an if/else for prepared value "items" and the only difference be the "blank" option.

Reply:
Maybe its ok to have the if/else have repeated selectOptions but one defines blank

Reply:
That is a very good idea that I didn't think of. I've changed it to an else clause.